### PR TITLE
Article about Web Workers and Comlink library

### PR DESCRIPTION
**Pull request by reramjiawan**
Article about Web Workers and Comlink library

### CODE REVIEW

File: data/blog/making-web-workers-enjoyable.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+---
+
+## Table of Contents
+
+<TOCInline toc={props.toc} exclude={["Table of Contents"]} toHeading={2} />
+
+## Background
+
+In the ever-evolving landscape of web development, the pursuit of creating responsive applications is a shared objective among developers.
+The aim is typically to achieve a seamless user experience by maintaining a minimum of 60 frames per second, a standard set by the highly responsive[^1] mobile apps that users have grown accustomed to.
```
Comment (DaveBitter) [inline data/blog/making-web-workers-enjoyable.md]: But the page responding to user input in general right? A choked up thread results in clicks and stuff not being instant which is arguably the most important

File: data/blog/making-web-workers-enjoyable.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+Comlink[^13] is a tiny library developed by Google. Its primary function is to simplify the process of working with Web Workers by eliminating the complexities associated with using `postMessage()`.
+It achieves this by adopting an RPC (Remote Procedure Call)[^14] style for message transmission and leveraging JavaScript Proxies[^15] that maintain a reference to the original target.
+In essence, Comlink enables seamless access to any element from the Main Thread within a Web Worker and vice versa.
+This bidirectional accessibility significantly enhances the developer experience.
+Furthermore when used together with TypeScript[^16] it supports autocomplete features, making coding even more efficient and enjoyable.
+
+### How to use Comlink
+
+Using Comlink is fairly straight forward as shown below:
```
Comment (DaveBitter) [inline data/blog/making-web-workers-enjoyable.md]: I feel like this bit could use a little more text telling what it is doing and how it works

File: data/blog/making-web-workers-enjoyable.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+In the ever-evolving landscape of web development, the pursuit of creating responsive applications is a shared objective among developers.
+The aim is typically to achieve a seamless user experience by maintaining a minimum of 60 frames per second, a standard set by the highly responsive[^1] mobile apps that users have grown accustomed to.
+However, this can often pose a significant challenge due to the single-threaded[^2] nature of JavaScript.
+
+In the current era, with an abundance of frontend libraries at our disposal, it’s increasingly easy to compromise on performance as these libraries consume substantial resources.
+This implies that when JavaScript is tasked with heavy computations or data processing, it can result in an unresponsive or janky user interface (UI), leading to user dissatisfaction.
+Therefore, it is crucial to manage resources effectively to ensure a smooth and responsive UI.
+
+## Web Workers
```
Comment (DaveBitter) [inline data/blog/making-web-workers-enjoyable.md:26]: Perhaps mention how these are different to service workers? I feel like they are often confused

**Review by DaveBitter — COMMENTED** [review]
Nice article! Didn't know about comlink yet :)

File: data/blog/making-web-workers-enjoyable.md (hunk 1 above)
Comment (reramjiawan) [reply data/blog/making-web-workers-enjoyable.md]: I have indeed mentioned it and its also mentioned in the paragraph about the idea behind Web Workers. Done.

File: data/blog/making-web-workers-enjoyable.md (hunk 2 above)
Comment (reramjiawan) [reply data/blog/making-web-workers-enjoyable.md]: Indeed was very little information. Done!

File: data/blog/making-web-workers-enjoyable.md (hunk 3 above)
Comment (reramjiawan) [reply data/blog/making-web-workers-enjoyable.md:26]: Done!

**Pull request merged**